app/components/page-components/Chatbot/module-bases/ChatbotModToolsBase.vue.ts

```diff
@@ -0,0 +1,273 @@
+  type: IListMetadata<string>;
+  duration: INumberMetadata;
+}
+
+interface IExcludedMetadata {
+  level: IListMetadata<number>;
+}
+
+interface IProtectionGeneralMetadata {
+  punishment: IPunishmentMetadata;
+  excluded: IExcludedMetadata;
+  message: ITextMetadata;
+}
+
+interface IProtectionAdvancedMetadata {
+  minimum: INumberMetadata;
+  maximum: INumberMetadata;
+  percent: ISliderMetadata;
+}
+
+interface ICapsProtectionMetadata {
+  general: IProtectionGeneralMetadata;
+  advanced: IProtectionAdvancedMetadata;
+}
+
+interface ISymbolProtectionMetadata {
+  general: IProtectionGeneralMetadata;
+  advanced: IProtectionAdvancedMetadata;
+};
+
+interface ILinkProtectionMetadata {
+  commands: {
+    permit: {
+      command: ITextMetadata;
+      description: ITextMetadata;
+      response: ITextMetadata;
+      response_type: IListMetadata<string>;
+      new_alias: ITextMetadata;
+    }
+  },
+  general: IProtectionGeneralMetadata;
+  new_whitelist_item: ITextMetadata;
+  new_blacklist_item: ITextMetadata;
+}
+
+interface IWordProtectionBlacklistItem {
+  text: ITextMetadata;
+  is_regex: IInputMetadata;
+  punishment: IPunishmentMetadata;
+}
+
+interface IWordProtectionMetadata {
+  general: IProtectionGeneralMetadata;
+  new_blacklist_item: IWordProtectionBlacklistItem;
+}
+
+interface IProtectionMetadata {
+  caps: ICapsProtectionMetadata;
+  symbol: ISymbolProtectionMetadata;
+  link: ILinkProtectionMetadata;
+  word: IWordProtectionMetadata;
+}
+
+
+@Component({
+  components: {
+    TextInput,
+    TextAreaInput,
+    ListInput,
+    NumberInput,
+    SliderInput
+  }
+})
+export default class ChatbotAlertsBase extends ChatbotWindowsBase {
+  capsProtection: ICapsProtectionData = null;
+  symbolProtection: ISymbolProtectionData = null;
+  linkProtection: ILinkProtectionData = null;
+  wordProtection: IWordProtectionData = null;
+
+  mounted() {
+    this.capsProtection = cloneDeep(this.capsProtectionResponse.settings);
+    this.symbolProtection = cloneDeep(this.symbolProtectionResponse.settings);
+    this.linkProtection = cloneDeep(this.linkProtectionResponse.settings);
+    this.wordProtection = cloneDeep(this.wordProtectionResponse.settings);
+  }
+
+  get capsProtectionResponse() {
+    return this.chatbotApiService.state.capsProtectionResponse;
+  }
+
+  get symbolProtectionResponse() {
+    return this.chatbotApiService.state.symbolProtectionResponse;
+  }
+
+  get linkProtectionResponse() {
+    return this.chatbotApiService.state.linkProtectionResponse;
+  }
+
+  get wordProtectionResponse() {
+    return this.chatbotApiService.state.wordProtectionResponse;
+  }
+
+  label(protectionType: string) {
+    switch (protectionType) {
+      case 'caps':
+        return 'Capitalized letters';
+      case 'symbol':
+        return 'Symbols';
+      case 'links':
+        return 'Links';
+      default:
+        return 'unpermitted value';
+    }
+  }
+
+  // metadata
+  generalMetadata(protectionType: string) {
```

Comment:
This method has a very weak type-checking. It allows returning any type of object here. There are some things you can do to improve that:

1) Add a returning type
`getMetadata(): Dictionary<IInputMetadata>`
2) Use `metadata` helper to have strong type protection
```
        duration: metadata.number({
          required: true,
          placeholder: $t('Punishment Duration in minutes'),
          min: 0
        })
```

Don't forget translations `$t()`